packages/widgets/src/lib/components/InferenceWidget/shared/inputValidation.ts

```diff
@@ -13,65 +14,76 @@ import type {
 	WidgetExampleZeroShotTextInput,
 } from "@huggingface/tasks";
 
-export function isTextInput<TOutput>(sample: WidgetExample<TOutput>): sample is WidgetExampleTextInput<TOutput> {
-	return "text" in sample;
+function isObject(arg: unknown): arg is Record<string, unknown> {
+	return !!arg && arg?.constructor === Object;
+}
+function isStrArray(arg: unknown): arg is string[] {
+	return Array.isArray(arg) && arg.every((v) => typeof v === "string");
+}
+
+export function isTextInput<TOutput>(sample: unknown): sample is WidgetExampleTextInput<TOutput> {
```

Comment:
Not really sure how it works, but sad that we had to go to `unknown` from `WidgetExample<TOutput>` :)